pandas/core/reshape/merge.py

```diff
@@ -1730,6 +1730,12 @@ def flip(xs) -> np.ndarray:
         )
         tolerance = self.tolerance
 
+        if is_object_dtype(right_values):
```

Comment:
right_values -> right_values.dtype

Reply:
Oh, I did not know that would work, thanks.